common/shell-local/run.go

```diff
@@ -169,17 +168,17 @@ func createFlattenedEnvVars(config *Config) (string, error) {
 	envVars["PACKER_BUILDER_TYPE"] = config.PackerBuilderType
 
 	// expose ip address variables
-	httpAddr := common.GetHTTPAddr()
-	if httpAddr != "" {
-		envVars["PACKER_HTTP_ADDR"] = httpAddr
+	httpAddr := config.generatedData["PackerHTTPAddr"]
+	if httpAddr != nil && httpAddr != "" {
+		envVars["PACKER_HTTP_ADDR"] = httpAddr.(string)
 	}
-	httpIP := common.GetHTTPIP()
-	if httpIP != "" {
-		envVars["PACKER_HTTP_IP"] = httpIP
+	httpIP := config.generatedData["PackerHTTPIP"]
+	if httpIP != nil && httpIP != "" {
+		envVars["PACKER_HTTP_IP"] = httpIP.(string)
 	}
-	httpPort := common.GetHTTPPort()
-	if httpPort != "" {
-		envVars["PACKER_HTTP_PORT"] = httpPort
+	httpPort := config.generatedData["PackerHTTPPort"]
+	if httpPort != nil && httpPort != "" {
+		envVars["PACKER_HTTP_PORT"] = httpPort.(string)
```

Comment:
I saw this being used as an int here: https://github.com/hashicorp/packer/pull/9238/files#diff-6af397a1ebe32399f6ec341ee8589609R52

Reply:
( ditto throughout )

Reply:
I can't see the diff the link is pointing 🤔

Reply:
I think @azr  is referring to https://github.com/hashicorp/packer/pull/9238/files#diff-0510d722beaf01f375fd8c2d964c8143R61

But if I understand the generatedData maps correctly this is the data after it has been passed into the provisioner so it may have started as an int but ends up as a string in the provision step. With that said I do see a few instances within the provisioners where `PACKER_HTTP_PORT` is overwritten with `common.GetHTTPPort()` does that need to be updated?

Reply:
Correct @nywilken!

Reply:
Yes ! Okay I got confused that we very often `Put("http_port", int)` and here it's a string !

Reply:
This can be confusing! I wanted to change to `Put("http_port", string)` but this is already being used as an `int` in a lot of other places so I decided to avoid changing it for now :|

Reply:
yes, that makes sense 🙂 👍